### PR TITLE
Bugfix/977 janky studio

### PR DESCRIPTION
**Pull request by kenjinp**
fixes: https://github.com/BlueBrain/nexus/issues/977

resolves jankiness by: 
- moving routing state to query params
- sorts workspace and dashboard results alphabetically, because the ordering is random, which causes jankiness when re-rendering tabs
- removes unnecessary re-rendering by moving components to be direct children of StudioView as a render function (see [here](https://github.com/BlueBrain/nexus-web/compare/Bugfix/977-janky-studio?expand=1#diff-ea6bdbcf23c5771fd39fc5aa667e45a1R38))

adds a `userQueryString` helper hook

### CODE REVIEW

File: src/shared/containers/DashboardListContainer.tsx [hunk 1]
```diff
@@ -26,11 +26,11 @@ const DashboardList: React.FunctionComponent<DashboardListProps> = ({
   orgLabel,
   projectLabel,
   workspaceId,
-  dashboardId,
   studioResourceId,
   refreshList,
 }) => {
-  const history = useHistory();
+  const [queryParams, setQueryString] = useQueryString();
```
Comment (Stafeeva) [inline src/shared/containers/DashboardListContainer.tsx]: like it! 👍

File: src/shared/containers/DashboardResultsContainer.tsx [hunk 2]
```diff
@@ -158,11 +136,12 @@ const DashboardResultsContainer: React.FunctionComponent<{
             return binding.self.value.includes(studioResourceId);
           });
         if (selectedResource === undefined && currentResource !== undefined) {
-          selectResource(currentResource.self.value, false);
+          selectResource(currentResource.self.value);
         }
         setItems(tempItems);
       })
       .catch(e => {
+        console.error(e);
```
Comment (Stafeeva) [inline src/shared/containers/DashboardResultsContainer.tsx]: do we want to log an error?

Comment (kenjinp) [reply src/shared/containers/DashboardResultsContainer.tsx]: probably not!

File: src/shared/containers/DashboardListContainer.tsx (hunk 1 above)
Comment (kenjinp) [reply src/shared/containers/DashboardListContainer.tsx]: 🦸🏼‍♂️

**Pull request merged**